src/DateTime.cc

```diff
@@ -201,6 +201,42 @@ NsType taiToUtc(NsType nsecs) {
     return nsecs - leapNSecs;
 }
 
+/** Convert a time in nsec from any time scale to TAI
+ * \param[in] nsecs Number of nanoseconds since the epoch in TAI
+ * \param[in] scale The time scale (TAI, TT or UTC)
+ * \return Number of nanoseconds since the epoch in TAI
+ */
+long long nsecAnyToTai(long long nsecs, dafBase::DateTime::Timescale scale) {
+    if (scale == dafBase::DateTime::TAI) {
+        return nsecs;
+    } else if (scale == dafBase::DateTime::TT) {
+        return nsecs - TT_MINUS_TAI_NSECS;
+    } else if (scale == dafBase::DateTime::UTC) {
+        return utcToTai(nsecs);
+    }
```

Comment:
Thank you for the explanation. I think I'll leave it, since the current code is very compact, but it's good to know about the enum warning for future reference.